### PR TITLE
Phoenix Regular Signals -> core

### PR DESCRIPTION
**Pull request by YHerm**

### CODE REVIEW

File: src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java [hunk 1]
```diff
@@ -0,0 +1,158 @@
...
+		) {
+			super(name, signal, slope, angleUnit);
+		}
+
+		public StatusSignal<Double> getSignalSlope() {
+			// For using refresh all with more signals...
+			return slopeSignal;
+		}
+
+	}
```
Comment (GilStein1) [inline src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java:156]: I don't think this should be a static class. Put it in a different file

File: src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java [hunk 2]
```diff
@@ -0,0 +1,158 @@
...
+import com.ctre.phoenix6.BaseStatusSignal;
+import com.ctre.phoenix6.StatusSignal;
+import frc.robot.hardware.signal.AngleSignal;
+import frc.robot.hardware.signal.DoubleSignal;
+import frc.utils.AngleUnit;
+import frc.utils.ctre.PhoenixProUtils;
+
+public class Phoenix6SignalBuilder {
+
+	private static final int UPDATE_FREQUENCY_RETRIES = 5;
```
Comment (GilStein1) [inline src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java:12]: constant?

File: src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+
+	public interface SignalGetter {
+
+		StatusSignal<Double> getSignal();
+
+	}
```
Comment (GilStein1) [inline src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java:18]: if you want to put all those classes in one file, then at least put them all at the bottom of the file

File: src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+    ) {
+		StatusSignal<Double> signalClone = cloneWithFrequency(signal, frequency);
+		setFrequencyWithRetry(signalSlope.getSignal(), frequency);
+		return new Phoenix6LatencySignal(signalClone.getName(), signalClone, signalSlope.getSignal(), angleUnit);
+	}
+	//@formatter:on
+
+	/**
+	 * Use only if not fetching the slope signal!!!
+	 */
```
Comment (GilStein1) [inline src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6SignalBuilder.java:55]: many comments

**Review by bossyshrimp68 — DISMISSED** [review]

File: src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6BothLatencySignal.java [hunk 5]
```diff
@@ -0,0 +1,23 @@
+package frc.robot.hardware.signal.phoenix;
+
+
```
Comment (drorta) [inline src/main/java/frc/robot/hardware/signal/phoenix/Phoenix6BothLatencySignal.java:3]: Intended?

**Review by drorta — APPROVED** [review]

**Review by hillelv — APPROVED** [review]

**Pull request merged**